### ISSUE TITLE
Flamingo: Fix unit tests (MockitoException at TypeCache.java)

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#5339]
**High-priority, so anyone feel free to send a pull request for this.
In other words, no need to ask to get assigned, everyone can start working on it immediately if they wish. :-)**

---

This branch is compatible with Flamingo (the latest Android studio):
https://github.com/commons-app/apps-android-commons/tree/v4.2.0-release (based on [Siva's branch](https://github.com/siva-subramaniam-v/apps-android-commons/tree/gradle-version-upgrade) )
And it runs well on devices.

Unfortunately, all unit tests fail with [this stacktrace](https://github.com/commons-app/apps-android-commons/actions/runs/6525905252/job/17719081619?pr=5220#step:10:271):
```
FAILED
    org.mockito.exceptions.base.MockitoException at TestCommonsApplication.kt:81
        Caused by: org.mockito.exceptions.base.MockitoException at TypeCache.java:152
            Caused by: java.lang.IllegalStateException at InlineBytecodeGenerator.java:177
                Caused by: java.lang.IllegalArgumentException at ClassReader.java:184
```
It happens since that branch switched to Flamingo and to a higher JDK level.
Updating `jacoco.gradle`'s `toolVersion` to `0.8.10` alone does not fix the unit tests, but it is probably worth doing.
In `app/build.gradle` I see that we are using 2.* mockito packages, while the latest versions seem to be 5.*, time to update maybe?
Some other packages may need to be updated too.
There is a possibility that some code changes may be needed.

Your task:
1. Pull from the branch above
2. Make the unit tests work (it is fine if a few still fail)
3. Send a pull request

To fix the unit tests, an idea could be to update version numbers of testing/mock-related packages. The latest stable version numbers can be seen on each project's website. Another idea could be searching for similar exceptions on Stack Overflow.

This is high priority because it prevents us from making our main branch Flamingo-friendly.
Looking forward to your pull requests. :-)

**Comment by nicolas-raoul** [comment]
@Sullynr2Wolf It is yours, thanks! Please let us know about your progress every day or so, if possible. :-)

**Comment by Sullynr2Wolf** [comment]
Day 1 update: Code and manual(and such) reading, understanding, and trying to locate the problem.
(As there are tons of assignments going on, it sure takes time for all those... Hopefully, I can figure out what's wrong.)|
Btw will there be a ddl or something I need to take into considerations?

**Comment by nicolas-raoul** [comment]
Sorry what is ddl? :-)

Have you tried updating version numbers as suggested in the description above? The latest stable version numbers can be seen on each project's website. I think that's the first thing to try. The second could be searching for similar exceptions on StackOverflow.

**Comment by Sullynr2Wolf** [comment]
@nicolas-raoul Sorry! By ddl I mean deadline, my apologies! I am still trying to set the Android Studios for now and it seems to be a little malfunctioning... I've upgraded the jacoco.gradle's toolVersion to 0.8.10, but it seems caused some catastrophe on my Android Studios, I am now still identifying the problem. Will get that done ASAP :)

**Comment by Sullynr2Wolf** [comment]
![image](https://github.com/commons-app/apps-android-commons/assets/141620882/8b3c91e9-5b74-4b90-b992-6f801b33e8b3)
Hi, I think right now I encountered this situation and I tried for a bit, but none of the method work.
Hope this question won't embarrass you too much... Sorry for such inconvenience. I am supposed to fix, not add more problems.

**Comment by Sullynr2Wolf** [comment]
I've been trying that a few times and this is not working.
Btw could it be that I set up the wrong SDK or JDK or Gradle with the wrong version or something like that?

**Comment by Sullynr2Wolf** [comment]
Hello! I tried that last night and found it was still the same problem after I reinstalled the Android Studio.
I asked in my course's forum and got feedback from my lecturer, who said this could be a little too hard for me.
![image](https://github.com/commons-app/apps-android-commons/assets/141620882/70a30a3a-b2ad-4712-bf6d-c8457366503b)
I will keep on trying though :) Thanks for the help! (If the issue is way too urgent, another volunteer could be really helpful as well!)
